h2o-py/h2o/backend/server.py

```diff
@@ -344,7 +345,10 @@ def _launch_server(self, port, baseport, mmax, mmin, ea, nthreads, jvm_custom_ar
         giveup_time = time.time() + self._TIME_TO_START
         while True:
             if proc.poll() is not None:
-                raise H2OServerError("Server process terminated with error code %d" % proc.returncode)
+                if proc.returncode == self._BAD_JAVA_VERSION_RETURN_CODE_:
+                    raise H2OServerError("Java version check failed: Only Java 8, 9, 10, 11 and 12 are supported.")
```

Comment:
I think this error message needs to say that the server failed to start or something along those lines as well

Reply:
fixed in 80d407c